Formula/zpaq.rb

```diff
@@ -1,27 +1,49 @@
 class Zpaq < Formula
   desc "Incremental, journaling command-line archiver"
   homepage "http://mattmahoney.net/dc/zpaq.html"
-  url "http://mattmahoney.net/dc/zpaq705.zip"
-  sha256 "d8abe3e3620d4c6f3ddc1da149acffa4c24296fd9c74c9d7b62319e308b63334"
-  version "7.05"
-
+  url "http://mattmahoney.net/dc/zpaq713.zip"
+  version "7.13"
+  sha256 "9120cf4fb1afdecea3ac4f690d7b0577f7cb004ca6b152856edd8ac444f0d919"
   head "https://github.com/zpaq/zpaq.git"
 
   bottle do
-    cellar :any_skip_relocation
-    sha256 "f0998826effe4dae407d041c4143a39b3645dc46a82456b4e86ac88f106babd3" => :el_capitan
-    sha256 "bdd4716595b8eed1c52d2a2ef43abb095a996d05a0489f81f0b22ae43a26d51d" => :yosemite
-    sha256 "a71ad4808da8c69844f72322ff7633b895d034028129b30c3d25409d91bd4213" => :mavericks
-    sha256 "a2d2215b51cc18370f5ebe700160d1d7143ce5d680b2365a5adb4bb1622d06fc" => :mountain_lion
+    cellar :any
+    sha256 "c75fda6498ab5c8535654bd1f8057b6fec6717bf62daab743e5fb0142bead204" => :el_capitan
   end
 
   def install
-    system "make"
-    include.install "libzpaq.h"
-    bin.install "zpaq"
+    # Makefile introduced for zpaq >= 7.06 was never adapted for OS X
+    # Reported 9th May 2016: https://github.com/zpaq/zpaq/pull/13
+    inreplace "Makefile" do |s|
+      # Use OS X style dylib names
+      s.gsub! "libzpaq.so.0.1", "libzpaq.0.1.dylib"
+      s.gsub! "libzpaq.so", "libzpaq.dylib"
+
+      # ld: unknown option: -soname
+      # ld: file not found: libzpaq.0.1.dylib
+      # use the correct arguments to compile a dylib
+      s.gsub! "-Wl,-soname,$(SONAME) -o $@ $<",
+              "-Wl -dynamiclib libzpaq.o zpaq.o -o $(SONAME)"
+
+      # clang: error: no such file or directory: 'zpaq.o'
+      # libzpaq.0.1.dylib needs both libzpaq.o and zpaq.o
+      s.gsub! "$(SONAME): libzpaq.o", "$(SONAME): libzpaq.o zpaq.o"
 
-    system "pod2man", "zpaq.pod", "zpaq.1"
-    man1.install "zpaq.1"
+      # OS X `install` command doesn't have `-t`
+      s.gsub! /(install -m.* )-t (.*) (.*)\r/, "\\1 \\3 \\2"
+
+      # Repair the corruption the prior replace just caused to the line endings
+      # Not necessary but may be useful
+      # s.gsub! /(.*)\r/, "\\1"
```

Comment:
Not a bug in `inreplace`, but a mistake in the preceding regular expression. The file being edited seems to have CRLF line endings, thus using `\r` as an end of line anchor in the preceding regular expression works. But because the `\r` is not included in the replacement pattern, lines affected by the substitution now only end in LF.

Most editors (and maybe `make`, too) are just fine with either all lines ending in CRLF or ending in LF, but you're creating a mix of CRLF and LF and so the interpretation ends up being that the file has LF line endings, causing the CR characters that are still present in some (most) lines to become visible.

Replacing

```ruby
s.gsub! /(install -m.* )-t (.*) (.*)\r/, "\\1 \\3 \\2"
```

with (or something similar)

```ruby
s.gsub! /(install -m.* )-t (.*) (.*)(\r)/, "\\1 \\3 \\2\\4"
```

should do the trick. (I'm not sure though, why the last two install arguments are reversed.) There's definitely some room for improvement left here …